packages/components/addon/components/hds/table/th-button-sort.hbs

```diff
@@ -0,0 +1,13 @@
+{{!
+  Copyright (c) HashiCorp, Inc.
+  SPDX-License-Identifier: MPL-2.0
+}}
+<button
+  type="button"
+  class="hds-table__th-button hds-table__th-button--sort"
+  {{on "click" this.onClick}}
+  aria-label="sort"
```

Comment:
This will potentially result in multiple buttons with the same label, which is confusing for screen reader users. Ideally, we would incorporate the column name (the `<th>` text content) so that we have something like 'sort by Artist' and 'sort by Release Year'.

Reply:
@MelSumner thoughts? I had a [thumbs up](https://hashicorp.slack.com/archives/C025N5V4PFZ/p1702049406587669?thread_ts=1702034833.897939&cid=C025N5V4PFZ) on this 🤔

One consideration: the information of the column's name/label is available only when the consumers provide the data via `@columns`, while when they use directly the `Table::ThSort` the content is yielded so we don't have access to it.

Reply:
the consumer can, in theory, to provide their own `aria-label`, so an alternative would be to address this through guidance

Reply:
> the consumer can, in theory, provide their own aria-label

if I understand correctly, this would mean that the consumer would have to provide an extra `aria-label` when used without `@columns` but not when used with `@columns`. I think having two different APIs for the two ways of using the table is not great DX. Can we maybe think of an alternative? (we can always ask consumers to add an `aria-label` in both cases, but this would mean that all the existing instances of the `Hds::Table` with sorting would need to be refactored (so, this would be a breaking change to release).